python/drned_xmnr/op/common_op.py

```diff
@@ -102,3 +110,66 @@ def perform(self):
         if self.filter.devcli_error is None:
             return {'success': 'Saved initial config.'}
         return {'failure': 'Could not save config.'}
+
+
+class ParseLogErrorsOp(ActionBase):
+    """ Action handler used to parse and print errors from drned-xmnr logfiles.
+    """
+    action_name = 'xmnr parse-log-errors'
+
+    def _init_params(self, params):
+        self.target_log = self.param_default(params, "target_log", None)
+
+    def perform(self):
+        return self.run_with_trans(self._perform, write=True)
+
+    def _perform(self, trans):
+        filepath = self._get_target_filepath()
+        try:
+            with open(filepath, 'r') as logfile:
+                problems = gather_problems(logfile)
+                self._store_problems(trans, problems)
```

Comment:
I would prefer that the transaction is open only for this call - it is not needed for `gather_problems`.

Reply:
right, parsing can unnecessarily extend transaction runtime, updated...